### ISSUE TITLE
icon was blur while switching input method in ibus

### ISSUE THREAD

**Issue by iravan** [Cangjians/ibus-cangjie#74]
Desktop environment : 

> Window Manager (xfwm4)
> Panel (xfce4-panel)
> Desktop Manager (xfdesktop)
> File Manager  (thunar)
> Session Manager (xfce4-session)
> Setting System (xfce4-settings)
> Application Finder (xfce4-appfinder)
> Settings Daemon (xfconf)

The blur icon of switching input method in ibus
![4](https://cloud.githubusercontent.com/assets/4957816/17462846/dafb3ffa-5ceb-11e6-8612-fd751e4e9709.png)

**Comment by bochecha** [comment]
Basically, our icon is bad, it's a quick thing we did with @yookoala back in the days.

However, both of us use GNOME, and GNOME doesn't use those iconds, so none of us has any incentive to fix this. (plus, none of us knows how to properly draw icons :wink: )

I'd be happy to receive a better icon as a patch, though.

**Comment by iravan** [comment]
I see the website has a icon. why not just use that : 
`http://cangjians.github.io/images/cangjie-with-codes.png`

**Comment by bochecha** [comment]
> I see the website has an icon. why not just use that

Because it's way too big, has too much details, and is super distracting with all the colours.

You don't just take a big picture and scale it down, that would make a horrible icon.

**Comment by iravan** [comment]
I could do that. Is that it better draw by myself ? I could do that little task.

**Comment by bochecha** [comment]
One problem is it needs to be readable on user panels, and those might be clear or dark.

For example, if you just make an icon with "倉" in a black color, it won't be readable on dark panels. Alternatively, if you make it in a white color, it won't be readable on clear panels.

Another issue is that it needs to be readable with panels of various sizes, since some desktop environments allow changing the size of the panel (this is probably the problem you encounter with the icon being blurry)

(Incidentally, since GNOME uses text instead of an icon, then the color of the « icon » is just the color of the font, and the size of the « icon » is just the size of the font, and themes can change both, and it avoids all this headache)

Basically, unless we can use an SVG which gets automatically scaled/colored by the desktop, I don't think there is any way to do something good here.

Maybe just look at what other IBus engines do, and do something like them?

**Comment by iravan** [comment]
ibus-table-* just make the foregroud color and background is transparent.
I think no need think too much. Just follow the way like others. make it work without issue as goal.
I think SVG is necessary as it supports scaling without issue and feel to export to other resolution. Flexible enough.

**Comment by iravan** [comment]
Hi , just draw icons for this project. Hope it helps. It's svg with high resolution in the zip file. you can export it to any size of image file from GIMP/inkscape. 
[cang and quick.zip](https://github.com/Cangjians/ibus-cangjie/files/409350/cang.and.quick.zip)
![text3383](https://cloud.githubusercontent.com/assets/4957816/17523676/0cfe8306-5e8e-11e6-81e8-dee3f2dc2de9.png)
![text3384](https://cloud.githubusercontent.com/assets/4957816/17523677/0cff4598-5e8e-11e6-8278-6e8138462aac.png)

**Comment by bochecha** [comment]
Will that be readable on a dark panel?

Can you try it and provide screenshots?

**Comment by iravan** [comment]
> Will that be readable on a dark panel? 
I think that depends how dark it was. For manjaro xfce's penal. It's panel color is #3c403e.
if the icon is what I posted - black. That the result to me is : 
The icon size is 20x20px : 
![11](https://cloud.githubusercontent.com/assets/4957816/17551182/26587436-5f2b-11e6-99e9-24e89ced8474.png)

But with white : 
![12](https://cloud.githubusercontent.com/assets/4957816/17551529/46632c92-5f2d-11e6-95e8-a6211a6b6e27.png)
![13](https://cloud.githubusercontent.com/assets/4957816/17551530/4668f4ce-5f2d-11e6-976a-773dcda5cdc1.png)

**Comment by iravan** [comment]
This is another quick icon that it seemed a better one.
![14](https://cloud.githubusercontent.com/assets/4957816/17552036/ff9757cc-5f2f-11e6-91c6-7b967cda08d8.png)
![15](https://cloud.githubusercontent.com/assets/4957816/17552116/7dcd351c-5f30-11e6-8b2a-77718bd56aca.png)
[quick2.svg.zip](https://github.com/Cangjians/ibus-cangjie/files/410920/quick2.svg.zip)

**Comment by iravan** [comment]
However , the icon in preference of ibus is using the same icon. I think it cannot use black or white as it usually conflict with panel and perference page of ibus. 
I think it better to use RED , BLUE , GREEN ... such color.
I prefer to use blue.   
Plus , the default ibus input method is "EN" which uses #415099 as it's foreground color.

**Comment by bochecha** [comment]
That « black on black » screenshot is completely unreadable.

The problem with making the icon white is that it would then become unreadable on a light-colored panel.

How do other IBus engines deal with this?

**Comment by iravan** [comment]
The other ibus engine's icon used "shadow" on it's icon pattern. 
I think we could either do the same say or add a background behind the icon.
e.g. Pattern of icon uses white. Add a dark solid circle or rectangle as it's background.

**Comment by bochecha** [comment]
> Pattern of icon uses white. Add a dark solid circle or rectangle as it's background. 

This is exactly what the current icons do.

**Comment by bochecha** [comment]
I think your problem is that we only provide a 16x16 icon, which is too smal for the ibus switcher.

Since we have the svg as well, maybe we should provide bigger icon sizes in addition to the 16x16 one?

**Comment by iravan** [comment]
I see. But is it the pattern I post qualified the needs ? 
If it is , let's keep going. 

Seems it's reasonable that adding a dark solid circle or rectangle as it's background.

I need the colour decision to make the icon. 
What colour of the pattern ? 
What colour of the dark solid circle or rectangle ?

**Comment by bochecha** [comment]
> Seems it's reasonable that adding a dark solid circle or rectangle as it's background.

Again, this is exactly what we already do!

**Comment by iravan** [comment]
> I think your problem is that we only provide a 16x16 icon, which is too smal for the ibus switcher.
> 
> Since we have the svg as well, maybe we should provide bigger icon sizes in addition to the 16x16 one?

I think yes.

**Comment by yookoala** [comment]
@iravan Have you installed the latest version of Ubuntu (17.10)? Do you still have this issue?

**Comment by mbridon** [comment]
I see no answer to @yookoala's question above in the past 7 years, so I doubt @iravan will answer anytime soon...

Let's close this then. Also, we moved everything away from Github and onto gitlab.freedesktop.org/cangjie/ibus-cangjie a while ago, if the issue still happens with the latest releases please open it there. :wink:
